src/Simulation/QSharpFoundation/Math/FloatingPoint.qs

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.Math {
+
+    /// # Summary
+    /// Returns a value that is not a number (i.e. NaN).
+    ///
+    /// # Ouputs
+    /// A double-precision floating point value that is not a number.
+    ///
+    /// # Remarks
+    /// The value output by this function follows IEEE 754 rules for how `NaN`
+    /// works when used with other double-precision floating point values.
+    /// For example, for any value `x` of type `Double`, `NaN() == x` is
+    /// `false`; this holds even if `x` is also `NaN()`.
+    ///
+    /// # See Also
+    /// - Microsoft.Quantum.Math.IsNaN
+    /// - Microsoft.Quantum.Math.IsInfinite
+    /// - Microsoft.Quantum.Math.IsFinite
+    function NaN() : Double {
+        return 0.0 / 0.0;
+    }
+
+    /// # Summary
+    /// Returns whether a given floating-point value is not a number (i.e.
+    /// is NaN).
+    ///
+    /// # Input
+    /// ## d
+    /// A floating-point value to be checked.
+    ///
+    /// # Output
+    /// `true` if and only if `d` is not a number.
+    ///
+    /// # Remarks
+    /// Since `NaN()` is the only floating-point value that does not equal
+    /// itself, this function should be used instead of checking conditions such
+    /// as `d = NaN()`.
+    ///
```

Comment:
I suspect you meant 
`d == NaN()` instead of 
`d = NaN()`.

Reply:
Yes, thanks for catching that typo!